frontend/workflows/k8s/src/scale-resources.tsx

```diff
@@ -0,0 +1,289 @@
+import { Wizard, WizardStep } from "@clutch-sh/wizard";
+import { string } from "yup";
+
+import type { ConfirmChild, ResolverChild, WorkflowProps } from ".";
+
+// Examples of valid quantities: 0.1, 100m, 128974848, 129e6, 129M,  128974848000m, 123Mi
+const QUANTITY_REGEX = /^([+-]?[0-9.]+)([eEinumkKMGTP]*[-+]?[0-9]*)$/;
+
+const DeploymentIdentifier: React.FC<ResolverChild> = ({ resolverType }) => {
+  const { onSubmit } = useWizardContext();
+  const deploymentData = useDataLayout("deploymentData");
+  const inputData = useDataLayout("inputData");
+
+  const onResolve = ({ results, input }) => {
+    // Decide how to process results.
+    deploymentData.assign(results[0]);
+    inputData.assign(input);
+    onSubmit();
+  };
+
+  return <Resolver type={resolverType} searchLimit={1} onResolve={onResolve} />;
+};
+
+const DeploymentDetails: React.FC<WizardChild> = () => {
+  const { onSubmit, onBack } = useWizardContext();
+  const deploymentData = useDataLayout("deploymentData");
+  const deployment = deploymentData.displayValue() as IClutch.k8s.v1.Deployment;
+  const update = (key: string, value: boolean) => {
+    deploymentData.updateData(key, value);
+  };
+
+  const currentDeploymentData = useDataLayout("currentDeploymentData");
+
+  const [containerName, setContainerName] = React.useState(
+    deployment.deploymentSpec.template.spec.containers[0].name
+  );
+
+  const [containerIndex, setContainerIndex] = React.useState(0);
+
+  React.useEffect(() => {
+    // save the original values of deployment spec
+    if (deployment) {
+      currentDeploymentData.assign(deployment);
+    }
+  }, []);
+
+  return (
+    <WizardStep error={deploymentData.error} isLoading={deploymentData.isLoading}>
+      <strong>Deployment Details</strong>
+      <MetadataTable
+        onUpdate={update}
+        data={[
+          { name: "Name", value: deployment.name },
+          { name: "Namespace", value: deployment.namespace },
+          {
+            name: "Container Name",
+            value: (
+              <Select
+                label="Container Name"
+                name="containerName"
+                onChange={value => {
+                  setContainerName(value);
+                  setContainerIndex(
+                    deployment.deploymentSpec.template.spec.containers.findIndex(
+                      container => container.name === value
+                    )
+                  );
+                  deploymentData.updateData("containerName", value);
+                }}
+                options={deployment.deploymentSpec.template.spec.containers.map(container => {
+                  return { label: container.name };
+                })}
+              />
+            ),
+          },
+          {
+            name: "CPU Limit",
+            value: deployment.deploymentSpec.template.spec.containers.find(
+              container => container.name === containerName
+            ).resources.limits.cpu,
```

Comment:
We're making this find request quite a few times, it might be worthwhile to pull this out into a const and just utilize that, that way we're not making the same request over and over. Looks like there's already a findContainer helper as well we could use.

ie:
```
const currentDeployment = deployment.deploymentSpec.template.spec.containers.find(container => container.name === containerName);
```

or 
```
const currentDeployment = findContainer({ deploymentSpec: deployment.deploymentSpec, containerName });
```

```suggestion
            value: currentDeployment.resources.limits.cpu,
```

Reply:
Updated.